cpp/tests/groupby/without_agg/groupby_wo_agg_test.cu

```diff
@@ -0,0 +1,793 @@
+    the_column->dtype_info = extra_info;
+
+    return the_column;
+  }
+
+
+  // Converts a tuple of host vectors into a vector of gdf_columns
+  std::vector<gdf_col_pointer>
+  initialize_gdf_columns(multi_column_t host_columns)
+  {
+    std::vector<gdf_col_pointer> gdf_columns;
+    convert_tuple_to_gdf_columns(gdf_columns, host_columns);
+    return gdf_columns;
+  }
+
+  /* --------------------------------------------------------------------------*/
+  /**
+   * @Synopsis  Initializes key columns and aggregation column for gdf group by call
+   *
+   * @Param key_count The number of unique keys
+   * @Param value_per_key The number of times a random aggregation value is generated for a key
+   * @Param max_key The maximum value of the key columns
+   * @Param max_val The maximum value of aggregation column
+   * @Param print Optionally print the keys and aggregation columns for debugging
+   */
+  /* ----------------------------------------------------------------------------*/
+  void create_input(const size_t key_count, const size_t value_per_key,
+                    const size_t max_key, const size_t max_val,
+                    bool print = false, const gdf_size_type n_count = 0) {
+    size_t shuffle_seed = rand();
+    initialize_keys(input_key, key_count, value_per_key, max_key, shuffle_seed);
+    initialize_values(input_value, key_count, value_per_key, max_val, shuffle_seed);
+
+    gdf_input_key_columns = initialize_gdf_columns(input_key);
+    gdf_input_value_column = create_gdf_column(input_value, n_count);
+
+    // Fill vector of raw pointers to gdf_columns
+    for(auto const& c : gdf_input_key_columns){
+      gdf_raw_input_key_columns.push_back(c.get());
+    }
+    gdf_raw_input_val_column = gdf_input_value_column.get();
+
+    if(print)
+    {
+      std::cout << "Key column(s) created. Size: " << std::get<0>(input_key).size() << std::endl;
+      print_tuple_vector(input_key);
+
+      std::cout << "Value column(s) created. Size: " << input_value.size() << std::endl;
+      print_vector(input_value); //todo@!
+
+      std::cout << "\n==============================================" << std::endl;
+
+    }
+  }
+
+    /* --------------------------------------------------------------------------*/
+    /**
+     * @Synopsis  Creates a unique_ptr that wraps a gdf_column structure intialized with a host vector
+     *
+     * @Param host_vector The host vector whose data is used to initialize the gdf_column
+     *
+     * @Returns A unique_ptr wrapping the new gdf_column
+     */
+    /* ----------------------------------------------------------------------------*/
+  // Compile time recursion to convert each vector in a tuple of vectors into
+  // a gdf_column and append it to a vector of gdf_columns
+  template<std::size_t I = 0, typename... Tp>
+  inline typename std::enable_if<I == sizeof...(Tp), void>::type
+  convert_tuple_to_gdf_columns(std::vector<gdf_col_pointer> &gdf_columns,std::tuple<std::vector<Tp>...>& t)
```

Comment:
Don't existing test utilities cover this functionality? Please use them.

Reply:
Please respond.

Reply:
Deleted it and used existing test utilities